sync/p2p_messages.md

```diff
@@ -80,14 +80,33 @@ Message is RLP encoded, fields:
 
 Message is RLP encoded, fields:
 
-  - `Port :: int` - listen port
-  - `Share :: int` - number of peers to share
+### Version 1
+
+  - `Port :: int` - listen port.
+  - `Share :: int` - number of peers to share.
+  - `GenesisHash :: byte_array`
+  - `Difficulty :: int` - the total difficulty of the chain.
+  - `TopHash :: byte_array`
+  - `SyncAllowed :: bool` - if the sender of this ping message is accepting
+    synchronization messages.
+  - `Peers :: [byte_array]` - list of shared peers.
+
+
+### Version 2
+
+  - `Versions :: list`
+    - `Protocol :: binary` - type of message.
+    - `Vsns :: [int]` - versions supported
+  - `Port :: int` - listen port.
+  - `Share :: int` - number of peers to share.
   - `GenesisHash :: byte_array`
+  - `Height :: int` - current height.
   - `Difficulty :: int` - the total difficulty of the chain.
   - `TopHash :: byte_array`
-  - `sync_allowed :: bool` - if the sender of this ping message is accepting
+  - `SyncAllowed :: bool` - if the sender of this ping message is accepting
```

Comment:
Since we have a capability field now, isn't it appropriate to move this there? Sounds like a capability to me.

Reply:
It could be, the current example capability is more complicated though, it is for a sync capability "chain_poi", which is:
```
{height, int}
{root_hash, binary}
{genesis, binary}
{top, binary}
{poi, binary} 
```
A downside would be that it increases the size of the ping.

Reply:
Okay, maybe a misunderstand "capabilities"

Reply:
I don't think we are sure that is how capabilities should be used, let's see when we actually find a use-case for them. My idea was that the capabilities should signal what _additional_ messages that the node will understand and react to 🤔

Reply:
Ulf wanted to use it for additional methods of sync. https://github.com/aeternity/aeternity/issues/4189